op-sync-tester/synctester/service.go

```diff
@@ -0,0 +1,246 @@
+	}
+	if err := s.initHTTPServer(cfg); err != nil {
+		return fmt.Errorf("failed to start HTTP server: %w", err)
+	}
+	return nil
+}
+
+func (s *Service) initMetrics(cfg *config.Config) {
+	if cfg.MetricsConfig.Enabled {
+		procName := "default"
+		s.metrics = metrics.NewMetrics(procName)
+		s.metrics.RecordInfo(cfg.Version)
+	} else {
+		s.metrics = metrics.NoopMetrics{}
+	}
+}
+
+func (s *Service) initPProf(cfg *config.Config) error {
+	s.pprofService = oppprof.New(
+		cfg.PprofConfig.ListenEnabled,
+		cfg.PprofConfig.ListenAddr,
+		cfg.PprofConfig.ListenPort,
+		cfg.PprofConfig.ProfileType,
+		cfg.PprofConfig.ProfileDir,
+		cfg.PprofConfig.ProfileFilename,
+	)
+
+	if err := s.pprofService.Start(); err != nil {
+		return fmt.Errorf("failed to start pprof service: %w", err)
+	}
+
+	return nil
+}
+
+func (s *Service) initMetricsServer(cfg *config.Config) error {
+	if !cfg.MetricsConfig.Enabled {
+		s.log.Info("Metrics disabled")
+		return nil
+	}
+	m, ok := s.metrics.(opmetrics.RegistryMetricer)
+	if !ok {
+		return fmt.Errorf("metrics were enabled, but metricer %T does not expose registry for metrics-server", s.metrics)
+	}
+	s.log.Debug("Starting metrics server", "addr", cfg.MetricsConfig.ListenAddr, "port", cfg.MetricsConfig.ListenPort)
+	metricsSrv, err := opmetrics.StartServer(m.Registry(), cfg.MetricsConfig.ListenAddr, cfg.MetricsConfig.ListenPort)
+	if err != nil {
+		return fmt.Errorf("failed to start metrics server: %w", err)
+	}
+	s.log.Info("Started metrics server", "addr", metricsSrv.Addr())
+	s.metricsSrv = metricsSrv
+	return nil
+}
+
+func (s *Service) initBackend(ctx context.Context, cfg *config.Config) error {
+	syncTesterCfg, err := cfg.SyncTester.Load(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to load sync tester config: %w", err)
+	}
+	b, err := backend.FromConfig(s.log, s.metrics, syncTesterCfg)
+	if err != nil {
+		return fmt.Errorf("failed to setup backend: %w", err)
+	}
+	s.backend = b
+	return nil
+}
+
+func (s *Service) initRPCHandler(cfg *config.Config) error {
+	s.rpcHandler = oprpc.NewHandler(cfg.Version,
+		oprpc.WithLogger(s.log),
+		oprpc.WithWebsocketEnabled(),
+	)
+	if cfg.RPC.EnableAdmin {
+		s.log.Info("Admin RPC enabled")
+		if err := s.rpcHandler.AddAPI(rpc.API{
+			Namespace:     "admin",
+			Service:       frontend.NewAdminFrontend(s.backend),
+			Authenticated: true,
+		}); err != nil {
+			return fmt.Errorf("failed to add admin API: %w", err)
+		}
+	}
+	if err := s.rpcHandler.AddRPC("/synctest"); err != nil {
+		return fmt.Errorf("failed to set up synctest route: %w", err)
+	}
+	if err := s.rpcHandler.AddAPIToRPC("/synctest", rpc.API{
+		Namespace: "sync",
+		Service:   frontend.NewSyncFrontend(s.backend),
+	}); err != nil {
+		return fmt.Errorf("failed to add sync API: %w", err)
+	}
+	return nil
+}
+
+func (s *Service) initHTTPServer(cfg *config.Config) error {
+	endpoint := net.JoinHostPort(cfg.RPC.ListenAddr, strconv.Itoa(cfg.RPC.ListenPort))
+	// middleware to parse http query
```

Comment:
Middleware to parse the query params and initialize a session. Session ID is temporally set to hash of these values. We may use a unique random id for the sessid to do parallel testing in the follow up PR.